File: Solutions/1249. Minimum Remove to Make Valid Parentheses/1249.java
class Solution {
    public String minRemoveToMakeValid(String st) {
        Stack<Integer> s = new Stack<>();
        char[] str = st.toCharArray();
        int n = str.length;
        for (int i=0;i<n;i++){
            if(str[i] == '('){
                s.push(i);
            }
            else if (str[i] == ')'){
                if(s.isEmpty()){
                    str[i]=' ';
                } else{
                    s.pop();
                }
            }
        }
        while(!s.isEmpty()){
            str[s.pop()] = ' ';
        }
        StringBuilder ans = new StringBuilder();
        for(int i=0;i<n;i++){
            if(str[i] != ' '){
                ans.append(str[i]);
            }
        }
        return new String(ans);
    }
}

File: Solutions/1249. Minimum Remove to Make Valid Parentheses/1249.py
class Solution(object):
    def minRemoveToMakeValid(self, s):
        """
        :type s: str
        :rtype: str
        """
        stack = []
        ans = list(s)
        for i in range(len(ans)):
            if s[i] == '(':
                stack.append(i)
            elif s[i] == ')':
                if len(stack) == 0:
                    ans[i] = ""
                else:
                    stack.pop()
        
        for i in stack:
            ans[i] = ""

        return ''.join(ans)

      